.github/workflows/build-push.yml

```diff
@@ -0,0 +1,84 @@
+# Name of the workflow
+name: build-push
+
+# This workflow will run when there is a push (including merge) to the listed branches
+on:
+  push:
+    branches:
+      - master
+      - v2.87
+
+# Variables available to all jobs
+env:
+  IMAGE_REPO: ${{ vars.DOCKERHUB_REPO }}
+  BUILD_NUMBER: ${{ github.run_number }}
+
+# Jobs that will run when the workflow is triggered
+jobs:
+  # This job will build and then push to docker hub
+  build-push:
+    # The type of runner the job will run on
+    runs-on: ubuntu-20.04
+
+    # Variables that are available to all steps in the job
+    env:
+      IMAGE_VERSION: ''
+    
+    steps:
+      # Upgrade Docker engine version, needed for building images.
+      - name: Install Latest Docker Version
+        run: |
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+          sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu  $(lsb_release -cs)  stable"
+          sudo apt-get update
+          sudo apt-get install docker-ce
+
+      # Authenticate Dockerhub to allow pushing to our image repo
+      - name: Login to Dockerhub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_TKN }}
+
+      # Checkout our Github repo
+      - name: Checkout Github Repo
+        uses: actions/checkout@v3
+
+      # Setup Scala
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v13
```

Comment:
who provides this action? We should document where it comes from and its source code repo.

Reply:
https://github.com/marketplace/actions/setup-scala
https://github.com/olafurpg/setup-scala

olafurpg/setup-scala has been deprecated as version 14. Recommendation from the author to move to coursier/setup-action.

https://github.com/marketplace/actions/coursier-setup-action
https://github.com/coursier/setup-action

Reply:
Thanks @naphelps 

Updated to use [https://github.com/coursier/setup-action](https://github.com/coursier/setup-action), comment added to document source of action.

Reply:
Also changed the unit test workflow to use the new setup-action instead of the deprecated one.